alvr/adb/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+// https://android.googlesource.com/platform/packages/modules/adb/+/refs/heads/main/docs/user/adb.1.md
+
+pub mod connection_state;
+pub mod device;
+pub mod forwarded_port;
+pub mod transport_type;
+
+use std::{io::Cursor, path::PathBuf, process::Command, time::Duration};
+
+use const_format::formatcp;
+use device::Device;
+use forwarded_port::ForwardedPort;
+use zip::ZipArchive;
+
+#[cfg(not(windows))]
+const ADB_EXECUTABLE: &str = "adb";
+#[cfg(windows)]
+const ADB_EXECUTABLE: &str = "adb.exe";
+
+// https://developer.android.com/tools/releases/platform-tools#revisions
+// NOTE: At the time of writing this comment, the revisions section above
+// shows the latest version as 35.0.2, but the latest that can be downloaded
+// by specifying a version is 35.0.0
+const PLATFORM_TOOLS_VERSION: &str = "-latest"; // E.g. "_r35.0.0"
+
+#[cfg(target_os = "linux")]
+const PLATFORM_TOOLS_OS: &str = "linux";
+#[cfg(target_os = "macos")]
+const PLATFORM_TOOLS_OS: &str = "darwin";
+#[cfg(windows)]
+const PLATFORM_TOOLS_OS: &str = "windows";
+
+const PLATFORM_TOOLS_URL: &str = formatcp!("https://dl.google.com/android/repository/platform-tools{PLATFORM_TOOLS_VERSION}-{PLATFORM_TOOLS_OS}.zip");
```

Comment:
I think here you should just use a non-const fn and `format!()` instead of `formatcp!()`

Reply:
because you don't have to use this in a const constext. So you can remove the const_format dependency altogether.